src/time.cr

```diff
@@ -605,10 +605,19 @@ struct Time
     {% end %}
 
     unless offset
-      # current TZ may have DST, either in past, present or future
-      ret = LibC.localtime_r(pointerof(second), out tm)
-      raise Errno.new("localtime_r") if ret.null?
-      offset = tm.tm_gmtoff.to_i64
+      {% if !flag?(:windows) %}
+        # current TZ may have DST, either in past, present or future
+        ret = LibC.localtime_r(pointerof(second), out tm)
+        raise Errno.new("localtime_r") if ret.null?
+        offset = tm.tm_gmtoff.to_i64
+      {% else %}
+        # TODO handle daylight?
+        # https://msdn.microsoft.com/en-us/library/windows/desktop/ms724421(v=vs.85).aspx
+        ret = LibWindows.get_time_zone_information(out tz)
+        raise WinError.new("GetTimeZoneInformation") if ret == -1
+        # UTC = local time + bias
+        return tz.bias.to_i64 * - Span::TicksPerMinute
+      {% end %}
```

Comment:
Pedantic: I usually use the reverse, specific first, and in alphabetical order:

```crystal
if flag?(:darwin)
  # macos specific
if flag?(:windows)
  # win specific
else
  # generic
end
```

As per [the documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724421(v=vs.85).aspx) I believe the following would work for DST:

```crystal
case ret
when LibWindows::TIME_ZONE_ID_UNKNOWN
  tz.bias.to_i64 * - Span::TicksPerMinute
when LibWindows::TIME_ZONE_ID_STANDARD
  tz.standard_bias.to_i64 * - Span::TicksPerMinute
when LibWindows::TIME_ZONE_ID_DAYLIGHT
  tz.daylight_bias.to_i64 * - Span::TicksPerMinute
end
```